### ISSUE TITLE
Add snaps API for adding tab to transaction approval

### ISSUE THREAD

**Issue by danfinlay** [MetaMask/metamask-plugin-beta#112]
We have had a recurring idea from a couple possible snaps developing teams. The type of teams include:

- Contract security auditors
- Smart contract analyzers
- Smart contract decompilers
- Metadata decoders
- Gas price analyzers

These use cases would like to add an extra tab to the transaction confirmation screen (currently `Details` and `Data`). This would allow services to add an extra layer of analysis to the view a person sees before a final confirmation.

Opening the topic here for further discussion.

**Comment by crazyrabbitLTC** [comment]
Yes, it would be nice if we could have a part of the MetaMask UI to use to communicate information via the plugin. Plugins are great, but in terms of the idea of having a security auditor plugin, it would be good to be able to have some space to see data about what is being shown. 

Also, the metamask UI might be a good location for plugings that have some sort of minimum settings required- for example login for decentralized storage, or something like that.

**Comment by crazyrabbitLTC** [comment]
I imagine this box could be identified by some sort of marker from Metamask saying something like, "This is a plugin, not Metamask" just to be clear.

**Comment by danfinlay** [comment]
To drill more carefully on the problem of labeling plugins within our UI, I've opened this issue: https://github.com/MetaMask/metamask-snaps-beta/issues/115
